omod/src/main/java/org/openmrs/module/adminui/page/controller/account/ChangeDefaultsPageController.java

```diff
@@ -0,0 +1,146 @@
+/**
+ * The contents of this file are subject to the OpenMRS Public License
+ * Version 1.0 (the "License"); you may not use this file except in
+ * compliance with the License. You may obtain a copy of the License at
+ * http://license.openmrs.org
+ *
+ * Software distributed under the License is distributed on an "AS IS"
+ * basis, WITHOUT WARRANTY OF ANY KIND, either express or implied. See the
+ * License for the specific language governing rights and limitations
+ * under the License.
+ *
+ * Copyright (C) OpenMRS, LLC.  All Rights Reserved.
+ */
+
+package org.openmrs.module.adminui.page.controller.account;
+
+import org.openmrs.User;
+import org.openmrs.api.UserService;
+import org.openmrs.api.context.Context;
+import org.openmrs.messagesource.MessageSourceService;
+import org.openmrs.module.adminui.AdminUiConstants;
+import org.openmrs.ui.framework.annotation.BindParams;
+import org.openmrs.ui.framework.annotation.MethodParam;
+import org.openmrs.ui.framework.annotation.SpringBean;
+import org.openmrs.ui.framework.page.PageModel;
+import org.openmrs.util.OpenmrsConstants;
+import org.springframework.context.MessageSource;
+import org.springframework.validation.BindingResult;
+import org.springframework.validation.ObjectError;
+
+import javax.servlet.http.HttpServletRequest;
+import java.util.List;
+import java.util.Map;
+
+public class ChangeDefaultsPageController {
+
+	public String get(PageModel pageModel) {
+		if (Context.isAuthenticated()) {
+			User user = Context.getAuthenticatedUser();
+			Map<String, String> props = user.getUserProperties();
+			UserDefaults userDefaults = new UserDefaults();
+			userDefaults.setDefaultLocale(props.get(OpenmrsConstants.USER_PROPERTY_DEFAULT_LOCALE));
+			userDefaults.setProficientLocales(props.get(OpenmrsConstants.USER_PROPERTY_PROFICIENT_LOCALES));
+			pageModel.addAttribute("userDefaults", userDefaults);
+			pageModel.addAttribute("locales", Context.getAdministrationService().getPresentationLocales());
+		}
+		return "account/changeDefaults";
+	}
+
+	public String post(@MethodParam("createUserDefaults") @BindParams UserDefaults defaults,
+	                   BindingResult errors,
+	                   @SpringBean("userService") UserService userService,
+	                   @SpringBean("messageSourceService") MessageSourceService messageSourceService,
+	                   @SpringBean("messageSource") MessageSource messageSource,
+	                   HttpServletRequest request,
+	                   PageModel model) {
+		if (errors.hasErrors()) {
+			sendErrorMessage(errors, messageSource, request);
+			model.addAttribute("errors", errors);
+			return "account/changePassword";
+		}
+		return saveDefaults(defaults, userService, messageSourceService, request);
+	}
+
+	private void sendErrorMessage(BindingResult errors, MessageSource messageSource, HttpServletRequest request) {
+		List<ObjectError> allErrors = errors.getAllErrors();
+		String message = getMessageErrors(messageSource, allErrors);
```

Comment:
The framework should be able to translate these codes for you, so you should not have to do this.